examples/cloud_export_example_test.go

```diff
@@ -1,85 +1,253 @@
 //go:build examples
 // +build examples
 
-//nolint:testpackage,forbidigo
+//nolint:forbidigo,goconst,testpackage
 package examples
 
 import (
 	"context"
 	"fmt"
 	"testing"
 
-	cloudexportpb "github.com/kentik/api-schema-public/gen/go/kentik/cloud_export/v202101beta1"
+	"github.com/AlekSi/pointer"
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestDemonstrateCloudExportAPI(t *testing.T) {
+func TestDemonstrateCloudExportAPIWithAWS(t *testing.T) {
 	t.Parallel()
-	err := demonstrateCloudExportAPI()
+	err := demonstrateCloudExportAPIWithAWS()
 	assert.NoError(t, err)
 }
 
-func demonstrateCloudExportAPI() error {
+func TestDemonstrateCloudExportAPIWithAzure(t *testing.T) {
+	t.Parallel()
+	err := demonstrateCloudExportAPIWithAzure()
+	assert.NoError(t, err)
+}
+
+func TestDemonstrateCloudExportAPIWithGCE(t *testing.T) {
+	t.Parallel()
+	err := demonstrateCloudExportAPIWithGCE()
+	assert.NoError(t, err)
+}
+
+func TestDemonstrateCloudExportAPIWithIBM(t *testing.T) {
+	t.Parallel()
+	err := demonstrateCloudExportAPIWithIBM()
+	assert.NoError(t, err)
+}
+
+func demonstrateCloudExportAPIWithAWS() error {
 	ctx := context.Background()
 	client, err := NewClient()
 	if err != nil {
 		return err
 	}
 
-	fmt.Println("Invoking client.CloudExportAdmin.ListCloudExport")
-	getAllResp, err := client.CloudExportAdmin.ListCloudExport(ctx, &cloudexportpb.ListCloudExportRequest{})
+	fmt.Println("Getting all cloud exports")
+	getAllResp, err := client.CloudExports.GetAll(ctx)
+	if err != nil {
+		return fmt.Errorf("client.CloudExports.GetAll: %w", err)
+	}
+
+	fmt.Println("Invalid cloud exports count:", getAllResp.InvalidCloudExportsCount)
+	fmt.Printf("Listed cloud exports: %+v\n", getAllResp.CloudExports)
+
+	fmt.Println("Creating AWS cloud export")
+	ce := models.NewAWSCloudExport(models.CloudExportAWSRequiredFields{
+		Name:          "example-aws-export",
+		PlanID:        "11467",
+		CloudProvider: models.CloudProviderAWS,
+		AWSProperties: models.AWSPropertiesRequiredFields{
+			Bucket: "dummy-bucket",
+		},
+	})
+	ce.Type = models.CloudExportTypeKentikManaged
+	ce.Enabled = pointer.ToBool(true)
+	ce.Description = "Dummy AWS description"
+	ce.APIRoot = "https://dummy.api.kentik.com"
+	ce.FlowDestination = "https://dummy.flow.kentik.com"
+	ce.AWSProperties.IAMRoleARN = "dummy-iam-role-arn"
+	ce.AWSProperties.Region = "dummy-region"
+	ce.AWSProperties.DeleteAfterRead = pointer.ToBool(true)
+	ce.AWSProperties.MultipleBuckets = pointer.ToBool(true)
+	ce.BGP = &models.BGPProperties{
```

Comment:
isn't it either AWS or BGP? I might be wrong but I thing they are alternatives and the one user is interested in is specified in CloudProvider field

Reply:
there should probably be a separate "demonstrateCloudExportAPIWithBGP"